lib/external/math.dart

```diff
@@ -40,4 +47,123 @@ ASTNode funcRand(Runtime runtime, ASTNode self, List<ASTNode> args) {
   }
 
   throw const UnexpectedTypeException('The rand method only takes [String, double, int, and bool] argument types');
+}
+
+ASTNode funcSin(Runtime runtime, ASTNode self, List<ASTNode> args){
+  runtimeExpectArgs(args, [ASTType.AST_ANY]);
+  final radian = args[0];
+  
+  switch(radian.type){
+    case ASTType.AST_INT: 
+    case ASTType.AST_DOUBLE: 
+      final DoubleNode result = DoubleNode()..doubleVal = sin(radian is DoubleNode ? radian.doubleVal : radian.intVal);
+      return result;
+    default: 
+  }
+  
+  throw const UnexpectedTypeException('The sin method only accept [double, int] argument types');
+}
+
+ASTNode funcCos(Runtime runtime, ASTNode self, List<ASTNode> args){
+  runtimeExpectArgs(args, [ASTType.AST_ANY]);
+  final radian = args[0];
+     
+  switch(radian.type){
+    case ASTType.AST_INT: 
+    case ASTType.AST_DOUBLE: 
+      final DoubleNode result = DoubleNode()..doubleVal = cos(radian is DoubleNode ? radian.doubleVal : radian.intVal);
+      return result;
+    default: 
+  }
+  
+  throw const UnexpectedTypeException('The cos method only accept [double, int] argument types');
+}
+
+ASTNode funcTan(Runtime runtime, ASTNode self, List<ASTNode> args){  
+  runtimeExpectArgs(args, [ASTType.AST_ANY]);
+  final radian = args[0];
+     
+  switch(radian.type){
+    case ASTType.AST_INT: 
+    case ASTType.AST_DOUBLE: 
+      final DoubleNode result = DoubleNode()..doubleVal = tan(radian is DoubleNode ? radian.doubleVal : radian.intVal);
+      return result;
+    default: 
+  }
+  
+  throw const UnexpectedTypeException('The tan method only accept [double, int] argument types');
```

Comment:
I do not think we need throw const here

Reply:
The rand function does throw const and it's not from me. So, I just follow along lol

Reply:
I mean const is optional

Reply:
It got `prefer_const_constructors` in **analysis_options.yaml** so Imma keep it for now